pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,240 @@
+package localclusterimport
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hive/v1/agent"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const LocalClusterNamespace = "local-cluster"
+
+type LocalClusterImport struct {
+	clusterImportOperations ClusterImportOperations
+	log                     *logrus.Logger
+}
+
+func NewLocalClusterImport(localClusterImportOperations ClusterImportOperations, log *logrus.Logger) LocalClusterImport {
+	return LocalClusterImport{clusterImportOperations: localClusterImportOperations, log: log}
+}
+
+func (i *LocalClusterImport) shouldImportLocalCluster() (bool, error) {
+	// The presence of agentClusterInstall for the local cluster will determine whether or not we need to perform an import
+	agentClusterInstall, err := i.clusterImportOperations.GetAgentClusterInstall(LocalClusterNamespace, LocalClusterNamespace+"-cluster-install")
+	if err != nil && !k8serrors.IsNotFound(err) {
+		i.log.Errorf("hubClusterDay2Importer: There was an error fetching the AgentClusterInstall while checking for need to register local cluster.")
+		return false, err
+	}
+	if agentClusterInstall != nil {
+		i.log.Infof("hubClusterDay2Importer: Found AgentClusterInstall for hub cluster, assuming that hub has been correctly registered for ZTP day 2 operations: %s", agentClusterInstall.Name)
+		return false, nil
+	}
+	return true, nil
+}
+
+func (i *LocalClusterImport) ImportLocalCluster() error {
+	clusterVersion, err := i.clusterImportOperations.GetClusterVersion("version")
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Unable to find cluster version info due to error: %s", err.Error())
+		return err
+	}
+
+	shouldImportLocalCluster, err := i.shouldImportLocalCluster()
+	if err != nil {
+		return err
+	}
+	if !shouldImportLocalCluster {
+		message := "hubClusterDay2Importer: no need to import local cluster as registration already detected"
+		i.log.Info(message)
+		return errors.New(message)
+	}
+	openshift_version := ""
+	release_image := ""
+	if clusterVersion != nil && clusterVersion.Status.History[0].State == configv1.CompletedUpdate {
+		openshift_version = clusterVersion.Status.History[0].Version
+		release_image = clusterVersion.Status.History[0].Image
+		i.log.Infof("hubClusterDay2Importer: Found openshift_version %s", openshift_version)
+	} else {
```

Comment:
suggesting to change the order of the if
```go
if clusterVersion == nil || clusterVersion.Status.History[0].State != configv1.CompletedUpdate {
		message := "hubClusterDay2Importer: Cluster version info is empty, cannot proceed"
		i.log.Error(message)
		return errors.New(message)
}
```
then you will not need the else and the code is clearer that way.